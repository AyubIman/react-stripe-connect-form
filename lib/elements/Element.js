import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Textbox } from 'weav';
import objectAssign from 'object-assign';

import { convertProperLabelFromKey } from '../utils/label';

import {
  textboxBasicStyle,
  labelBasicStyle,
} from './Element.style';

class Element extends Component {
  constructor(props) {
    super(props);

    const {
      value,
    } = props;

    this.state = {
      value,
    };

    this.updateElement = this.updateElement.bind(this);
  }

  updateElement(e) {
    const {
      objectPath,
      onValueChange,
    } = this.props;
    const { value } = e.target;

    this.setState({
      value,
    });

    onValueChange(objectPath, value);
  }

  render() {
    const {
      label,
      specialLabels,
      fontSize,
      labelStyle,
      textboxStyle,
    } = this.props;

    const {
      value,
    } = this.state;

    const id = `id_${Math.random()}`;

    return (
      <div>
        <p
          htmlFor={id}
          style={
            objectAssign(
              labelBasicStyle,
              {
                fontSize,
              },
              labelStyle,
            )
          }
        >
          {convertProperLabelFromKey(label, specialLabels)}
        </p>
        <Textbox
          value={value}
          id={id}
          style={
            objectAssign(
              textboxBasicStyle,
              textboxStyle,
            )
          }
          onChange={this.updateElement}
        />
      </div>
    );
  }
}

Element.propTypes = {
  label: PropTypes.string,
  // eslint-disable-next-line react/forbid-prop-types
  value: PropTypes.any,
  // eslint-disable-next-line react/forbid-prop-types
  specialLabels: PropTypes.object,
  fontSize: PropTypes.number.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  textboxStyle: PropTypes.object,
  // eslint-disable-next-line react/forbid-prop-types
  labelStyle: PropTypes.object,
  onValueChange: PropTypes.func.isRequired,
  objectPath: PropTypes.string,
};

Element.defaultProps = {
  label: '',
  value: '',
  specialLabels: {},
  textboxStyle: {},
  labelStyle: {},
  objectPath: '',
};

export default Element;
