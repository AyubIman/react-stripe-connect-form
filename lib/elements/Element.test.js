/* eslint-disable */
// test file
import React from 'react';
import { shallow } from 'enzyme';
// , mount, render

import Element from './Element';

describe('Element', () => {
  test('Shallow render of the element. Basic render, no props.', () => {
    const wrapper = shallow(
      <Element
        onValueChange={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the element. Render with value.', () => {
    const wrapper = shallow(
      <Element
        value="Foobar"
        onValueChange={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the element. Render with value and label.', () => {
    const wrapper = shallow(
      <Element
        label="Foo"
        value="Bar"
        onValueChange={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the element with specific font size.', () => {
    const wrapper = shallow(
      <Element
        label="Foo"
        fontSize={12}
        value="Bar"
        onValueChange={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the element with special textbox style.', () => {
    const wrapper = shallow(
      <Element
        label="Foo"
        textboxStyle={{
          color: 'red',
        }}
        value="Bar"
        onValueChange={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the element with special label style.', () => {
    const wrapper = shallow(
      <Element
        label="Foo"
        labelStyle={{
          color: 'red',
        }}
        value="Bar"
        onValueChange={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the element with special labels.', () => {
    const wrapper = shallow(
      <Element
        label="Foo"
        specialLabels={{
          Foo: "Bar",
        }}
        value="Bar"
        onValueChange={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
