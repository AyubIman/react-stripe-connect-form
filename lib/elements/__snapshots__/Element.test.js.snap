// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element Shallow render of the element with special label style. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Element
    fontSize={12}
    label="Foo"
    labelStyle={
      Object {
        "color": "red",
      }
    }
    objectPath=""
    onValueChange={[Function]}
    specialLabels={Object {}}
    textboxStyle={Object {}}
    value="Bar"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          htmlFor="id_0.9858485626028084"
          style={
            Object {
              "color": "red",
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            }
          }
        >
          FOO
        </p>,
        <Textbox
          id="id_0.9858485626028084"
          maxLength={null}
          onChange={[Function]}
          onClickOutside={[Function]}
          placeholder=""
          style={
            Object {
              "fontFamily": "sans-serif",
            }
          }
          textboxType="text"
          value="Bar"
          wrapperStyle={Object {}}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FOO",
          "htmlFor": "id_0.9858485626028084",
          "style": Object {
            "color": "red",
            "fontFamily": "sans-serif",
            "fontSize": 12,
            "marginBottom": 0,
            "marginTop": 0,
            "paddingTop": 10,
          },
        },
        "ref": null,
        "rendered": "FOO",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "id_0.9858485626028084",
          "maxLength": null,
          "onChange": [Function],
          "onClickOutside": [Function],
          "placeholder": "",
          "style": Object {
            "fontFamily": "sans-serif",
          },
          "textboxType": "text",
          "value": "Bar",
          "wrapperStyle": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            htmlFor="id_0.9858485626028084"
            style={
              Object {
                "color": "red",
                "fontFamily": "sans-serif",
                "fontSize": 12,
                "marginBottom": 0,
                "marginTop": 0,
                "paddingTop": 10,
              }
            }
          >
            FOO
          </p>,
          <Textbox
            id="id_0.9858485626028084"
            maxLength={null}
            onChange={[Function]}
            onClickOutside={[Function]}
            placeholder=""
            style={
              Object {
                "fontFamily": "sans-serif",
              }
            }
            textboxType="text"
            value="Bar"
            wrapperStyle={Object {}}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FOO",
            "htmlFor": "id_0.9858485626028084",
            "style": Object {
              "color": "red",
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            },
          },
          "ref": null,
          "rendered": "FOO",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "id_0.9858485626028084",
            "maxLength": null,
            "onChange": [Function],
            "onClickOutside": [Function],
            "placeholder": "",
            "style": Object {
              "fontFamily": "sans-serif",
            },
            "textboxType": "text",
            "value": "Bar",
            "wrapperStyle": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Element Shallow render of the element with special labels. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Element
    fontSize={12}
    label="Foo"
    labelStyle={Object {}}
    objectPath=""
    onValueChange={[Function]}
    specialLabels={
      Object {
        "Foo": "Bar",
      }
    }
    textboxStyle={Object {}}
    value="Bar"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          htmlFor="id_0.04382965776956915"
          style={
            Object {
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            }
          }
        >
          Bar
        </p>,
        <Textbox
          id="id_0.04382965776956915"
          maxLength={null}
          onChange={[Function]}
          onClickOutside={[Function]}
          placeholder=""
          style={
            Object {
              "fontFamily": "sans-serif",
            }
          }
          textboxType="text"
          value="Bar"
          wrapperStyle={Object {}}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bar",
          "htmlFor": "id_0.04382965776956915",
          "style": Object {
            "fontFamily": "sans-serif",
            "fontSize": 12,
            "marginBottom": 0,
            "marginTop": 0,
            "paddingTop": 10,
          },
        },
        "ref": null,
        "rendered": "Bar",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "id_0.04382965776956915",
          "maxLength": null,
          "onChange": [Function],
          "onClickOutside": [Function],
          "placeholder": "",
          "style": Object {
            "fontFamily": "sans-serif",
          },
          "textboxType": "text",
          "value": "Bar",
          "wrapperStyle": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            htmlFor="id_0.04382965776956915"
            style={
              Object {
                "fontFamily": "sans-serif",
                "fontSize": 12,
                "marginBottom": 0,
                "marginTop": 0,
                "paddingTop": 10,
              }
            }
          >
            Bar
          </p>,
          <Textbox
            id="id_0.04382965776956915"
            maxLength={null}
            onChange={[Function]}
            onClickOutside={[Function]}
            placeholder=""
            style={
              Object {
                "fontFamily": "sans-serif",
              }
            }
            textboxType="text"
            value="Bar"
            wrapperStyle={Object {}}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bar",
            "htmlFor": "id_0.04382965776956915",
            "style": Object {
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            },
          },
          "ref": null,
          "rendered": "Bar",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "id_0.04382965776956915",
            "maxLength": null,
            "onChange": [Function],
            "onClickOutside": [Function],
            "placeholder": "",
            "style": Object {
              "fontFamily": "sans-serif",
            },
            "textboxType": "text",
            "value": "Bar",
            "wrapperStyle": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Element Shallow render of the element with special textbox style. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Element
    fontSize={12}
    label="Foo"
    labelStyle={Object {}}
    objectPath=""
    onValueChange={[Function]}
    specialLabels={Object {}}
    textboxStyle={
      Object {
        "color": "red",
      }
    }
    value="Bar"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          htmlFor="id_0.12499727400439786"
          style={
            Object {
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            }
          }
        >
          FOO
        </p>,
        <Textbox
          id="id_0.12499727400439786"
          maxLength={null}
          onChange={[Function]}
          onClickOutside={[Function]}
          placeholder=""
          style={
            Object {
              "color": "red",
              "fontFamily": "sans-serif",
            }
          }
          textboxType="text"
          value="Bar"
          wrapperStyle={Object {}}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FOO",
          "htmlFor": "id_0.12499727400439786",
          "style": Object {
            "fontFamily": "sans-serif",
            "fontSize": 12,
            "marginBottom": 0,
            "marginTop": 0,
            "paddingTop": 10,
          },
        },
        "ref": null,
        "rendered": "FOO",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "id_0.12499727400439786",
          "maxLength": null,
          "onChange": [Function],
          "onClickOutside": [Function],
          "placeholder": "",
          "style": Object {
            "color": "red",
            "fontFamily": "sans-serif",
          },
          "textboxType": "text",
          "value": "Bar",
          "wrapperStyle": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            htmlFor="id_0.12499727400439786"
            style={
              Object {
                "fontFamily": "sans-serif",
                "fontSize": 12,
                "marginBottom": 0,
                "marginTop": 0,
                "paddingTop": 10,
              }
            }
          >
            FOO
          </p>,
          <Textbox
            id="id_0.12499727400439786"
            maxLength={null}
            onChange={[Function]}
            onClickOutside={[Function]}
            placeholder=""
            style={
              Object {
                "color": "red",
                "fontFamily": "sans-serif",
              }
            }
            textboxType="text"
            value="Bar"
            wrapperStyle={Object {}}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FOO",
            "htmlFor": "id_0.12499727400439786",
            "style": Object {
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            },
          },
          "ref": null,
          "rendered": "FOO",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "id_0.12499727400439786",
            "maxLength": null,
            "onChange": [Function],
            "onClickOutside": [Function],
            "placeholder": "",
            "style": Object {
              "color": "red",
              "fontFamily": "sans-serif",
            },
            "textboxType": "text",
            "value": "Bar",
            "wrapperStyle": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Element Shallow render of the element with specific font size. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Element
    fontSize={12}
    label="Foo"
    labelStyle={Object {}}
    objectPath=""
    onValueChange={[Function]}
    specialLabels={Object {}}
    textboxStyle={Object {}}
    value="Bar"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          htmlFor="id_0.26775032666966836"
          style={
            Object {
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            }
          }
        >
          FOO
        </p>,
        <Textbox
          id="id_0.26775032666966836"
          maxLength={null}
          onChange={[Function]}
          onClickOutside={[Function]}
          placeholder=""
          style={
            Object {
              "fontFamily": "sans-serif",
            }
          }
          textboxType="text"
          value="Bar"
          wrapperStyle={Object {}}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FOO",
          "htmlFor": "id_0.26775032666966836",
          "style": Object {
            "fontFamily": "sans-serif",
            "fontSize": 12,
            "marginBottom": 0,
            "marginTop": 0,
            "paddingTop": 10,
          },
        },
        "ref": null,
        "rendered": "FOO",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "id_0.26775032666966836",
          "maxLength": null,
          "onChange": [Function],
          "onClickOutside": [Function],
          "placeholder": "",
          "style": Object {
            "fontFamily": "sans-serif",
          },
          "textboxType": "text",
          "value": "Bar",
          "wrapperStyle": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            htmlFor="id_0.26775032666966836"
            style={
              Object {
                "fontFamily": "sans-serif",
                "fontSize": 12,
                "marginBottom": 0,
                "marginTop": 0,
                "paddingTop": 10,
              }
            }
          >
            FOO
          </p>,
          <Textbox
            id="id_0.26775032666966836"
            maxLength={null}
            onChange={[Function]}
            onClickOutside={[Function]}
            placeholder=""
            style={
              Object {
                "fontFamily": "sans-serif",
              }
            }
            textboxType="text"
            value="Bar"
            wrapperStyle={Object {}}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FOO",
            "htmlFor": "id_0.26775032666966836",
            "style": Object {
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            },
          },
          "ref": null,
          "rendered": "FOO",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "id_0.26775032666966836",
            "maxLength": null,
            "onChange": [Function],
            "onClickOutside": [Function],
            "placeholder": "",
            "style": Object {
              "fontFamily": "sans-serif",
            },
            "textboxType": "text",
            "value": "Bar",
            "wrapperStyle": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Element Shallow render of the element. Basic render, no props. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Element
    fontSize={12}
    label=""
    labelStyle={Object {}}
    objectPath=""
    onValueChange={[Function]}
    specialLabels={Object {}}
    textboxStyle={Object {}}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          htmlFor="id_0.48191843027333814"
          style={
            Object {
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            }
          }
        >
          
        </p>,
        <Textbox
          id="id_0.48191843027333814"
          maxLength={null}
          onChange={[Function]}
          onClickOutside={[Function]}
          placeholder=""
          style={
            Object {
              "fontFamily": "sans-serif",
            }
          }
          textboxType="text"
          value=""
          wrapperStyle={Object {}}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "htmlFor": "id_0.48191843027333814",
          "style": Object {
            "fontFamily": "sans-serif",
            "fontSize": 12,
            "marginBottom": 0,
            "marginTop": 0,
            "paddingTop": 10,
          },
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "id_0.48191843027333814",
          "maxLength": null,
          "onChange": [Function],
          "onClickOutside": [Function],
          "placeholder": "",
          "style": Object {
            "fontFamily": "sans-serif",
          },
          "textboxType": "text",
          "value": "",
          "wrapperStyle": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            htmlFor="id_0.48191843027333814"
            style={
              Object {
                "fontFamily": "sans-serif",
                "fontSize": 12,
                "marginBottom": 0,
                "marginTop": 0,
                "paddingTop": 10,
              }
            }
          >
            
          </p>,
          <Textbox
            id="id_0.48191843027333814"
            maxLength={null}
            onChange={[Function]}
            onClickOutside={[Function]}
            placeholder=""
            style={
              Object {
                "fontFamily": "sans-serif",
              }
            }
            textboxType="text"
            value=""
            wrapperStyle={Object {}}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "htmlFor": "id_0.48191843027333814",
            "style": Object {
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            },
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "id_0.48191843027333814",
            "maxLength": null,
            "onChange": [Function],
            "onClickOutside": [Function],
            "placeholder": "",
            "style": Object {
              "fontFamily": "sans-serif",
            },
            "textboxType": "text",
            "value": "",
            "wrapperStyle": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Element Shallow render of the element. Render with value and label. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Element
    fontSize={12}
    label="Foo"
    labelStyle={Object {}}
    objectPath=""
    onValueChange={[Function]}
    specialLabels={Object {}}
    textboxStyle={Object {}}
    value="Bar"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          htmlFor="id_0.009015355644049983"
          style={
            Object {
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            }
          }
        >
          FOO
        </p>,
        <Textbox
          id="id_0.009015355644049983"
          maxLength={null}
          onChange={[Function]}
          onClickOutside={[Function]}
          placeholder=""
          style={
            Object {
              "fontFamily": "sans-serif",
            }
          }
          textboxType="text"
          value="Bar"
          wrapperStyle={Object {}}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FOO",
          "htmlFor": "id_0.009015355644049983",
          "style": Object {
            "fontFamily": "sans-serif",
            "fontSize": 12,
            "marginBottom": 0,
            "marginTop": 0,
            "paddingTop": 10,
          },
        },
        "ref": null,
        "rendered": "FOO",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "id_0.009015355644049983",
          "maxLength": null,
          "onChange": [Function],
          "onClickOutside": [Function],
          "placeholder": "",
          "style": Object {
            "fontFamily": "sans-serif",
          },
          "textboxType": "text",
          "value": "Bar",
          "wrapperStyle": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            htmlFor="id_0.009015355644049983"
            style={
              Object {
                "fontFamily": "sans-serif",
                "fontSize": 12,
                "marginBottom": 0,
                "marginTop": 0,
                "paddingTop": 10,
              }
            }
          >
            FOO
          </p>,
          <Textbox
            id="id_0.009015355644049983"
            maxLength={null}
            onChange={[Function]}
            onClickOutside={[Function]}
            placeholder=""
            style={
              Object {
                "fontFamily": "sans-serif",
              }
            }
            textboxType="text"
            value="Bar"
            wrapperStyle={Object {}}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FOO",
            "htmlFor": "id_0.009015355644049983",
            "style": Object {
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            },
          },
          "ref": null,
          "rendered": "FOO",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "id_0.009015355644049983",
            "maxLength": null,
            "onChange": [Function],
            "onClickOutside": [Function],
            "placeholder": "",
            "style": Object {
              "fontFamily": "sans-serif",
            },
            "textboxType": "text",
            "value": "Bar",
            "wrapperStyle": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Element Shallow render of the element. Render with value. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Element
    fontSize={12}
    label=""
    labelStyle={Object {}}
    objectPath=""
    onValueChange={[Function]}
    specialLabels={Object {}}
    textboxStyle={Object {}}
    value="Foobar"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          htmlFor="id_0.20947825382147056"
          style={
            Object {
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            }
          }
        >
          
        </p>,
        <Textbox
          id="id_0.20947825382147056"
          maxLength={null}
          onChange={[Function]}
          onClickOutside={[Function]}
          placeholder=""
          style={
            Object {
              "fontFamily": "sans-serif",
            }
          }
          textboxType="text"
          value="Foobar"
          wrapperStyle={Object {}}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "htmlFor": "id_0.20947825382147056",
          "style": Object {
            "fontFamily": "sans-serif",
            "fontSize": 12,
            "marginBottom": 0,
            "marginTop": 0,
            "paddingTop": 10,
          },
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "id_0.20947825382147056",
          "maxLength": null,
          "onChange": [Function],
          "onClickOutside": [Function],
          "placeholder": "",
          "style": Object {
            "fontFamily": "sans-serif",
          },
          "textboxType": "text",
          "value": "Foobar",
          "wrapperStyle": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            htmlFor="id_0.20947825382147056"
            style={
              Object {
                "fontFamily": "sans-serif",
                "fontSize": 12,
                "marginBottom": 0,
                "marginTop": 0,
                "paddingTop": 10,
              }
            }
          >
            
          </p>,
          <Textbox
            id="id_0.20947825382147056"
            maxLength={null}
            onChange={[Function]}
            onClickOutside={[Function]}
            placeholder=""
            style={
              Object {
                "fontFamily": "sans-serif",
              }
            }
            textboxType="text"
            value="Foobar"
            wrapperStyle={Object {}}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "htmlFor": "id_0.20947825382147056",
            "style": Object {
              "fontFamily": "sans-serif",
              "fontSize": 12,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingTop": 10,
            },
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "id_0.20947825382147056",
            "maxLength": null,
            "onChange": [Function],
            "onClickOutside": [Function],
            "placeholder": "",
            "style": Object {
              "fontFamily": "sans-serif",
            },
            "textboxType": "text",
            "value": "Foobar",
            "wrapperStyle": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
